### PR TITLE
Submission from external source rejected with 403

### LINKED ISSUE #3023: A user with submission rights over a collection cannot import from an external source
**Describe the bug**
A non-admin user with submission rights over a collection cannot import from an external source due to a permission error.

**To Reproduce**
1. Login with user dspacedemo+submit@gmail.com
2. Go to MySpace -> Import metadata from an external source
3. Search from an external source (Ex. Pubmed)
4. Select an item and start a submission 
5. Choose a collection on which the user has submission permission (ex. Publications 2)

IssueScreenshot

**Expected behavior**
The workspaceitem should be created with a non-admin user with permission rights over the collection.

### PR DESCRIPTION
**Pull request by alemarte**
## References
* Fixes #3023

## Description
Solves the referenced issue.

## Instructions for Reviewers
The authorization check inside [ExternalDataServiceImpl](https://github.com/4Science/DSpace/blob/57417884acd97c36f4485e7d83ab8fe22be04451/dspace-api/src/main/java/org/dspace/external/service/impl/ExternalDataServiceImpl.java#L94) has been removed since the check is performed again later on during the [workspaceitem creation](https://github.com/4Science/DSpace/blob/57417884acd97c36f4485e7d83ab8fe22be04451/dspace-api/src/main/java/org/dspace/content/WorkspaceItemServiceImpl.java#L85).
After this modification the case described inside the issue on how to reproduce the behaviour results as fixed (and the implemented integration test doesn't fail as well).

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java [hunk 1]
```diff
@@ -3829,6 +3829,55 @@ public void createWorkspaceItemFromExternalSourcesUnauthorized() throws Exceptio
...
+                                     .contentType(parseMediaType(
+                                         TEXT_URI_LIST_VALUE))
+                                     .content("https://localhost:8080/server/api/integration/externalsources/" +
+                                                  "mock/entryValues/one"))
+                                    .andExpect(status().isCreated()).andReturn();
+
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        workspaceItemId = (Integer) map.get("id");
+        String itemUuidString = String.valueOf(((Map) ((Map) map.get("_embedded")).get("item")).get("uuid"));
```
Comment (tdonohue) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java:3864]: @alemarte : This entire section of code can be simplified if you'd just use `andDo()`, similar to other `post` calls in this same IT class, e.g. https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java#L837-L842

The code you have here will "work", but it's much slower as you are loading the entire result into a Map in order to just get the `id` returned.  Instead, you should be able to do something like
```
AtomicReference<Integer> idRef = new AtomicReference<>();
getClient(token).perform(post("/api/submission/workspaceitems")
                    .param("owningCollection", col1.getID().toString())
                    .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
                    .content("https://localhost:8080/server/api/integration/externalsources/" +
                                                  "mock/entryValues/one"))
                    .andExpect(status().isCreated())
                    .andExcept(jsonPath("$._embedded.collection.id", is(col1.getID().toString())))
                    .andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "$.id")));
workspaceItemId = idRef.get();
```

We also should be *cleaning up* in this method, by having a `try()` / `finally`, with `WorkspaceItemBuilder.deleteWorkspaceItem(idRef.get());` in the `finally` block.

This change in code does mean that you won't be able to get the `itemUuidString`, but I don't think that's necessary for this test.  All you need to ensure is that the WorkspaceItem is created in the correct collection, having *metadata* which matches the mock external source.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@alemarte : Overall this PR looks good. Thank you!  However, I have some suggestions on how to  improve your Integration Test (see inline below) in order to make it faster & also ensure it cleans up any data it creates (which is a best practice).

Once this PR is updated, I'd be in favor of merging.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java (hunk 1 above)
Comment (alemarte) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java]: Thanks for your review @tdonohue. All the suggested improvements have been committed.

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now!  Thanks @alemarte !

**Pull request merged**